state/unit.go

```diff
@@ -1713,7 +1715,10 @@ func (u *Unit) assignToMachineOps(m *Machine, unused bool) ([]txn.Op, error) {
 			{{"machineid", m.Id()}},
 		},
 	}}...)
-	massert := isAliveDoc
+	massert := append(isAliveDoc, bson.D{{
```

Comment:
Can we have a test for this.
Also, there needs to be business logic above which check that the machine has the HostUnits job or else this will result in a "state changing too quickly error. A unit test would have highlighted this issue.

Reply:
Sure thing